Replicator/c4RemoteReplicator.hh

```diff
@@ -220,6 +223,12 @@ namespace litecore {
       private:
         alloc_slice const      _url;
         const C4SocketFactory* _socketFactory{nullptr};
+        // _socketExternalKey comes from C4ReplicatorParameters::externalKey. This parameter is
+        // part of kC4ReplicatorOptionAuthentication. When kC4ReplicatorAuthClientCertKeyIsExternal is true, we
+        // don't have the private key data in kC4ReplicatorAuthClientCertKey. Instead, the private key will
+        // be in the form of object, externalKey.
+        std::unique_ptr<C4KeyPair, void (*)(C4KeyPair*)> _socketExternalKey{nullptr,
+                                                                            [](C4KeyPair* k) { c4keypair_release(k); }};
```

Comment:
One thing I have noticed in `C4Cert` is that the base declaration of the `C4Cert` struct, including its inheritance is outside `COUCHBASE_ENTERPRISE`. If doing the same for C4KeyPair, will it solve the problem here?

```
struct C4Cert final
    : public fleece::RefCounted
    , public fleece::InstanceCountedIn<C4Cert>
    , C4Base {
#ifdef COUCHBASE_ENTERPRISE
...
#endif  // COUCHBASE_ENTERPRISE
```